tests/test_trainer_distributed.py

```diff
@@ -57,9 +45,28 @@ def forward(self, input_ids, labels=None):
                 return input_ids
 
 
+class TestTrainerDistributed(TestCasePlus):
+    @require_torch_multigpu
+    def test_trainer(self):
+
+        output_dir = self.get_auto_remove_tmp_dir()
+        argv = f"--output_dir {output_dir}".split()
+
+        distributed_args = f"-m torch.distributed.launch --nproc_per_node={torch.cuda.device_count()} {self.test_file_dir}/test_trainer_distributed.py".split()
+        cmd = [sys.executable] + distributed_args + argv
+        result = execute_async_std(cmd, env=self.get_env(), stdin=None, timeout=180, quiet=False, echo=False)
+
+        assert result.stdout, "produced no output"
+        if result.returncode > 0:
+            pytest.fail(f"failed with returncode {result.returncode}")
+
+
 if __name__ == "__main__":
+    # The script below is meant to be run in torch.distributed, on a machine with multiple GPUs:
```

Comment:
Let me know if you wanted most of the original notes here, I nuked most of it, but perhaps the non-multigpu notes are still useful. The part I nuked starts at https://github.com/huggingface/transformers/blob/master/tests/test_trainer_distributed.py#L8